source/Coop.Core/Server/Services/Clans/Handler/ServerClanHandler.cs

```diff
@@ -0,0 +1,183 @@
+﻿using Common.Logging;
+using Common.Messaging;
+using Common.Network;
+using Coop.Core.Client.Services.Clans.Messages;
+using Coop.Core.Server.Services.Clans.Messages;
+using GameInterface.Services.Clans.Messages;
+using Serilog;
+using System;
+
+namespace Coop.Core.Server.Services.Clans.Handler
+{
+    /// <summary>
+    /// Handles all changes to clans on server.
+    /// </summary>
+    public class ServerClanHandler : IHandler
+    {
+        private readonly IMessageBroker messageBroker;
+        private readonly INetwork network;
+        private readonly ILogger Logger = LogManager.GetLogger<ServerClanHandler>();
+
+        public ServerClanHandler(IMessageBroker messageBroker, INetwork network)
+        {
+            this.messageBroker = messageBroker;
+            this.network = network;
+            messageBroker.Subscribe<NetworkClanNameChangeRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkClanLeaderChangeRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkClanKingdomChangeRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkDestroyClanRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkAddCompanionRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkAddRenownRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkChangeClanInfluenceRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkAdoptHeroRequest>(Handle);
+
+            messageBroker.Subscribe<NetworkNewHeirRequest>(Handle);
```

Comment:
I highly recommend splitting these up into separate handlers under the `Clan` service